src/FFT.c

```diff
@@ -0,0 +1,111 @@
+#include "FFT.h"
+#include <math.h>
+#include <stdlib.h>
+
+/**
+ * This method implements bit reversal needed by the FFT.
+ * Uses the Gold and Rader's bit reversal algorithm
+ * @param length The length of the input vectors. Must be a power of 2
+ * @param realArray 1D array containing the real part of the incoming vector.
+ * @param imaginaryArray 1D array containing the imaginary part of the incoming vector.
+ */
+static void bitReversal(unsigned length, fft_real* realArray, fft_real* imaginaryArray) {
+  
+  const unsigned N2 = length >> 1;
+  unsigned j = 0;
+  for (unsigned i = 0; i < length - 1; ++i) {
+    if (i < j) {
+      fft_real tmpReal = realArray[i];
```

Comment:
The tmpReal and tmpImaginary are not something that we want here. The goal of using temporary variables is to store the content of the memory in a register for faster access in the future. In that case, we are only accessing those variables once, so we need a few more instructions to mov the values from memory to registers. This means that we have 1 access to memory and 1 access to registers at line 22. If we remove the tmp variables we end up using only 1 access to memory. 

Fun fact, since we're not compiling with optimizations, the compiler won't even store those values in a register. It will simply store a copy somewhere else in memory. This means that using temporary variables will never end up giving better performances.  https://godbolt.org/z/WdEvjWhr1

Reply:
I need them for swapping